metapool/sample_sheet.py

```diff
@@ -1044,10 +1046,27 @@ def make_sample_sheet(metadata, table, sequencer, lanes, strict=True):
         sheet._add_metadata_to_sheet(metadata, sequencer)
         sheet._add_data_to_sheet(table, sequencer, lanes, metadata['Assay'],
                                  strict)
-        return sheet
-    else:
-        for message in messages:
-            message.echo()
+
+        # now that we have a SampleSheet() object, validate it for any
+        # additional errors that may have been present in the data and/or
+        # metadata.
+        messages = sheet.quiet_validate_and_scrub_sample_sheet()
+
+        if not any([isinstance(m, ErrorMessage) for m in messages]):
+            # No error messages equals success.
+            return sheet
+
+    # Continue legacy behavior of echoing ErrorMessages and WarningMessages.
+    msgs = []
+    for message in messages:
+        msgs.append(str(message))
+        message.echo()
+
+    # Introduce an exception raised for API calls that aren't reporting echo()
```

Comment:
Just to confirm, this will not follow legacy behavior, right? Mainly because it will actually raise a ValueError vs. just echo it, no?

Reply:
Yes. However, since it's only used by Jupyter notebooks and users of this module, it's not expected to break existing code.